consensus/hotstuff/timeoutcollector/aggregation.go

```diff
@@ -158,10 +169,19 @@ func (a *TimeoutSignatureAggregator) UnsafeAggregate() ([]flow.Identifier, []byt
 		//  * empty `signatures` slice, i.e. sharesNum == 0, which we exclude by earlier check
 		//  * if some signature(s), included via TrustedAdd, could not be decoded
 		if crypto.IsInvalidInputsError(err) {
-			return nil, nil, model.NewInvalidSignatureIncludedErrorf("signatures with invalid structure were included via TrustedAdd: %w", err)
+			return nil, nil, nil, model.NewInvalidSignatureIncludedErrorf("signatures with invalid structure were included via TrustedAdd: %w", err)
 		}
-		return nil, nil, fmt.Errorf("BLS signature aggregation failed: %w", err)
+		return nil, nil, nil, fmt.Errorf("BLS signature aggregation failed: %w", err)
+	}
+
+	valid, err := crypto.VerifyBLSSignatureManyMessages(pks, aggSignature, messages, hashers)
```

Comment:
As @AlexHentschel explained here, if the module does not support `TrustedAdd` anymore, the final verification isn't needed. The aggregated signature is guaranteed to be valid because adding and verifying are now always coupled. The purpose of `VerifyAndAdd` is to remove this final expensive check.

Reply:
I would have the extra verification to be 100% sure that we are dealing with correct sig, maybe we could remove that later ? This gives strong guarantee that produced signature is always valid and node won't be slashed. So API safety as priority

Reply:
In the current implementation (with only `VerifyAndAdd` updating the internal set), there is no risk of the aggregated signature being invalid. API safety checks should only be added if there is a risk the API caller can misuse the available functions. This is not the case here because the aggregated signature is valid by construction. A vague analogy is to write a code that computes `sum(a,b) = a+b` but then add a check `sum(a,b) - a == b` for safety. The check isn't necessary and correctness of the implementation should be maintained by unit tests. 

However, I agree with you that there is nothing in our code that guarantees no one will add a `TrustedAdd` function in the future 😅  I don't think Go can enforce only `VerifyAndAdd` can touch the internal set. If that happens then the extra check becomes required.

We usually don't add checks of code correctness within the implementation itself (that's what tests rather do), however I let you and @AlexHentschel decide if you want to keep the extra check. From a cryptographic point of view, that check is not needed here.

Reply:
The final verification of the aggregate is particularly expensive in this case, as the signatures are over different messages. Therefore, my preference would be to skip the posterior check, as it impacts the hot-path in consensus.